devdoc/constbuffer_array_splitter_requirements.md

```diff
@@ -58,3 +62,38 @@ This takes the `buffers` from a `CONSTBUFFER_ARRAY_HANDLE` and splits them into
 **SRS_CONSTBUFFER_ARRAY_SPLITTER_42_016: [** `constbuffer_array_splitter_split` shall succeed and return the `split_buffers`. **]**
 
 **SRS_CONSTBUFFER_ARRAY_SPLITTER_42_017: [** If there are any other failures then `constbuffer_array_splitter_split` shall fail and return `NULL`. **]**
+
+### constbuffer_array_splitter_split_to_array_of_array
+
+```c
+MOCKABLE_FUNCTION(, CONSTBUFFER_ARRAY_HANDLE*, constbuffer_array_splitter_split_to_array_of_array, CONSTBUFFER_ARRAY_HANDLE, buffers, uint32_t, max_buffer_size);
+```
+
+This takes the `buffers` from a `CONSTBUFFER_ARRAY_HANDLE` and splits them into a new `CONSTBUFFER_ARRAY_HANDLE*` where each `CONSTBUFFER_ARRAY_HANDLE` is of size `max_buffer_size` (or smaller for the final remaining buffer). The caller is responsible for cleaning up the returned `CONSTBUFFER_ARRAY_HANDLE*`.
+
+If `buffers` is `NULL` then `constbuffer_array_splitter_split_to_array_of_array` shall fail and return `NULL`.
+
+If `max_buffer_size` is `0` then `constbuffer_array_splitter_split_to_array_of_array` shall fail and return `NULL`.
+
+`constbuffer_array_splitter_split_to_array_of_array` shall call `constbuffer_array_get_buffer_count`.
+
+If the buffer count is 0 then `constbuffer_array_splitter_split_to_array_of_array` shall call `constbuffer_array_create_empty` and return the result.
```

Comment:
This isn't exactly correct because we need to return an array of `CONSTBUFFER_ARRAY_HANDLE`.